Add new test cases: mariadb_ssl

STARRY-S opened the pull request:
## New test cases: mariadb_ssl

- Setup mariadb multi-machine testing environment.
- Try to connect the mariadb server with the '--ssl' parameter.

----

- Related ticket: https://progress.opensuse.org/issues/109154
- Needles: NA
- Verification run: 
  - leap 15.4 x64 server: https://openqa.opensuse.org/tests/2272582
  - leap 15.4 x64 client : https://openqa.opensuse.org/tests/2272583
  - TW x64 client : https://openqa.opensuse.org/tests/2272624
  - TW x64 server: https://openqa.opensuse.org/tests/2272623
  - Leap 15.4 aarch64 client: https://openqa.opensuse.org/tests/2275513
  - Leap 15.4 aarch64 server:https://openqa.opensuse.org/tests/2275512
  
    ---------

  - sle15sp4 x64 server: https://openqa.suse.de/tests/8480225
  - sle15sp4 x64 client  : https://openqa.suse.de/tests/8480226
  - sle15sp4 aarch64 server: https://openqa.suse.de/tests/8480227
  - sle15sp4 aarch64 client  : https://openqa.suse.de/tests/8480228
  - sle15sp4 s390x server: https://openqa.suse.de/tests/8480229
  - sle15sp4 s390x client  : https://openqa.suse.de/tests/8480230

jouyingbin commented:
@STARRY-S 
Could you please 
1. Rename the case as **mariadb_ssl**
2. Is it possible to have a SLES VR
Thankyou.

STARRY-S commented:
> @STARRY-S Could you please
> 
> 1. Rename the case as **mariadb_ssl**
> 2. Is it possible to have a SLES VR  (If not, it's no problem too)
>    Thankyou.

Hi Ben, thanks for your comments,
1. I've updated the test code and renamed `mariadb` to `mariadb_ssl`.
2. I can not connect to OSD currently, so I will run it on OSD later,
    (maybe next week).

Amrysliu commented on schedule/security/mariadb_ssl.yaml:
```diff
@@ -0,0 +1,20 @@
+description:    >
+    mariadb test for sle, Leap and Tumbleweed
+schedule:
+    - boot/boot_to_desktop
```
If this case run on s390, you need to load test module `bootloader_zkvm`

STARRY-S replied:
Thanks, I'll update it.

Amrysliu reviewed (approved):
lgtm

rfan1 commented:
Starry,

For s390x mm tests, one more comment, did you use the same ip addr as CC test code?

lilyeyes commented:
In "Verification run": s/sle14sp4/sle15sp4/

STARRY-S commented:
> Starry,
> 
> For s390x mm tests, one more comment, did you use the same ip addr as CC test code?

Hi Richard, thanks for your comments,
I updated the YAML job templates to add the `SERVER_IP` and `CLIENT_IP` settings for the s390x platform,
and here are the VRs:
<https://openqa.suse.de/tests/8480370>
<https://openqa.suse.de/tests/8480371>

STARRY-S commented:
> In "Verification run": s/sle14sp4/sle15sp4/

Thanks, I've fixed it.

lilyeyes commented on tests/security/mariadb/mariadb_ssl_client.pm:
```diff
@@ -0,0 +1,41 @@
+#
+# Summary: Run mariadb connect to server with '--ssl' parameter test case
+# Maintainer: Starry Wang <starry.wang@suse.com> Ben Chou <bchou@suse.com>
+# Tags: poo#109154
```
Please add the tc#$test-case-number if have

STARRY-S replied:
Thanks, added.

lilyeyes commented on tests/security/mariadb/mariadb_ssl_server.pm:
```diff
@@ -0,0 +1,55 @@
+#
+# Summary: Setup mariadb server enviroment and wait client to connect
+# Maintainer: Starry Wang <starry.wang@suse.com> Ben Chou <bchou@suse.com>
+# Tags: poo#109154
```
Please add the tc#$test-case-number if have

STARRY-S replied:
Thanks, added.

lilyeyes commented on tests/security/mariadb/mariadb_ssl_client.pm:
```diff
@@ -0,0 +1,41 @@
+
+sub run {
+    my ($self) = @_;
+    select_console 'root-console';
```
Please move "select_console 'root-console';" out of the "parameters definition area" and put it into bellow "code area".

STARRY-S replied:
Done.

lilyeyes commented on tests/security/mariadb/mariadb_ssl_server.pm:
```diff
@@ -0,0 +1,55 @@
+
+sub run {
+    my ($self) = @_;
+    select_console 'root-console';
```
Please move "select_console 'root-console';" out of the "parameters definition area" and put it into bellow "code area".

STARRY-S replied:
Done.

STARRY-S commented:
> Starry,
> 
> For s390x mm tests, one more comment, did you use the same ip addr as CC test code?

Hi Richard, thanks for your comments.
I updated the YAML job templates to set `CLIENT_IP` and `SERVER_IP` for s390x.
And here are the verification runs:
https://openqa.suse.de/tests/8480401
https://openqa.suse.de/tests/8480400

rfan1 commented:
> > Starry,
> > For s390x mm tests, one more comment, did you use the same ip addr as CC test code?
> 
> Hi Richard, thanks for your comments. I updated the YAML job templates to set `CLIENT_IP` and `SERVER_IP` for s390x. And here are the verification runs: https://openqa.suse.de/tests/8480401 https://openqa.suse.de/tests/8480400

Great

rfan1 reviewed (approved):
LGTM

lilyeyes reviewed (approved):
LGTM.

jouyingbin commented:
Great work and LGTM besides small typo in git description.

- Setup mariadb multi-machine testing **enviroment**.

STARRY-S commented:
> Great work and LGTM besides small typo in git description.
> 
> * Setup mariadb multi-machine testing **enviroment**.

Thanks, I've fixed it.

jouyingbin reviewed (approved):
LGTM. thanks for the case.
